cogs/moderation.py

```diff
@@ -129,26 +129,44 @@ async def note(self, ctx: commands.Context):
         """ 
         raise NotImplementedError('Command requires implementation and permission set-up.')
     
-    @commands.hybrid_group()
+    @commands.hybrid_group(invoke_without_command=False)
     async def purge(self, ctx: commands.Context):
         """
         Commands for bulk deletion.
         """
-        raise NotImplementedError('Command requires implementation and permission set-up.')
+        pass
     
     @purge.command()
-    async def message(self, ctx: commands.Context):
+    async def message(self, ctx: commands.Context, number: int):
         """
         Bulk delete messages.
         """ 
-        raise NotImplementedError('Command requires implementation and permission set-up.')
+        async for message in ctx.channel.history(limit=number):
+            try:
+                await message.delete()
+            except discord.Forbidden:
+                await ctx.reply('I do not have permission to delete messages.', delete_after=5)
+                return
+            except discord.HTTPException:
+                await ctx.reply('An error occurred while deleting messages.', delete_after=5)
+                return
+        await ctx.reply(f'Deleted {number} messages.', delete_after=5)
     
     @purge.command()
-    async def reaction(self, ctx: commands.Context):
+    async def reaction(self, ctx: commands.Context, number: int):
```

Comment:
This is not the correct implementation I believe. The command should take in a specific message (i.e. a channel, then a message in the channel) and then also the reaction to purge from it.